src/agent/application.cpp

```diff
@@ -315,10 +323,20 @@ void Application::DeinitRcpMode(void)
 #endif
 }
 
+void Application::CreateNcpMode(void)
+{
+    otbr::Host::NcpHost &ncpHost = static_cast<otbr::Host::NcpHost &>(mHost);
```

Comment:
nit: Could we add a common method `GetNcpHost()` to get the ` otbr::Host::NcpHost`.

Reply:
I prefer not adding this because the `mHost` can be RcpHost or NcpHost. Calling the method in wrong cases will cause a fatal error. So let's try to avoid the cast as possible. Finally we want to make the RcpHost/NcpHost transparent to the code here.